datadotworld/client/_swagger/apis/sql_api.py

```diff
@@ -0,0 +1,285 @@
+        >>> thread = api.sql_get_with_http_info(owner, id, query, callback=callback_function)
+
+        :param callback function: The callback function
+            for asynchronous request. (optional)
+        :param str owner: User name and unique identifier of the creator of the dataset. For example, in the URL: [https://data.world/jonloyens/an-intro-to-dataworld-dataset](https://data.world/jonloyens/an-intro-to-dataworld-dataset), jonloyens is the unique identifier of the dataset. (required)
+        :param str id: Dataset unique identifier.   For example, in the URL: [https://data.world/jonloyens/an-intro-to-dataworld-dataset](https://data.world/jonloyens/an-intro-to-dataworld-dataset), an-intro-to-dataworld-dataset is the unique identifier of the dataset. (required)
+        :param str query: (required)
+        :param bool include_table_schema: Flags indicating to include table schema in the response.
+        :return: None
+                 If the method is called asynchronously,
+                 returns the request thread.
+        """
+
+        all_params = ['owner', 'id', 'query', 'include_table_schema']
+        all_params.append('callback')
+        all_params.append('_return_http_data_only')
+        all_params.append('_preload_content')
+        all_params.append('_request_timeout')
+
+        params = locals()
+        for key, val in iteritems(params['kwargs']):
+            if key not in all_params:
+                raise TypeError(
+                    "Got an unexpected keyword argument '%s'"
+                    " to method sql_get" % key
+                )
+            params[key] = val
+        del params['kwargs']
+        # verify the required parameter 'owner' is set
+        if ('owner' not in params) or (params['owner'] is None):
+            raise ValueError("Missing the required parameter `owner` when calling `sql_get`")
+        # verify the required parameter 'id' is set
+        if ('id' not in params) or (params['id'] is None):
+            raise ValueError("Missing the required parameter `id` when calling `sql_get`")
+        # verify the required parameter 'query' is set
+        if ('query' not in params) or (params['query'] is None):
+            raise ValueError("Missing the required parameter `query` when calling `sql_get`")
+
+
+        collection_formats = {}
+
+        path_params = {}
+        if 'owner' in params:
+            path_params['owner'] = params['owner']
+        if 'id' in params:
+            path_params['id'] = params['id']
+
+        query_params = []
+        if 'query' in params:
+            query_params.append(('query', params['query']))
+        if 'include_table_schema' in params:
+            query_params.append(('includeTableSchema', params['include_table_schema']))
+
+        header_params = {}
+
+        form_params = []
+        local_var_files = {}
+
+        body_params = None
+        # HTTP header `Accept`
+        header_params['Accept'] = self.api_client.\
```

Comment:
It is a problem that the auto-generated code doesn't allow overriding headers. In this case, I would prefer to give users choices of formats for query results. How could we accomplish that? (same comment applies to `sparql`)

Reply:
(Bringing in what I mentioned during standup in here). As regards our discussion on possible ways to resolve hardcoded Accept headers, subclassing the `APIClient` to override `select_header_accept` would have been a possible solution but we might still need to tweak the api endpoint for `sql` and `sparql` so that their `kwargs` can expect `accept_header` as key or maybe another param that allows the user add other Accept header.